service/src/client.ts

```diff
@@ -259,9 +259,15 @@ export class Client extends EventEmitter {
       return this._invokeDeviceMethod(deviceId, moduleIdOrMethodParams, methodParamsOrDone, done);
     }
 
-    return tripleValueCallbackToPromise((_callback) => {
-      this._invokeDeviceMethod(deviceId, moduleIdOrMethodParams, methodParamsOrDone, _callback);
-    }, (r: any, m: IncomingMessage) => { return createResultWithIncomingMessage(r, m); }, callback);
+    if (typeof moduleIdOrMethodParams === 'string') {
+      return tripleValueCallbackToPromise((_callback) => {
+        this._invokeDeviceMethod(deviceId, moduleIdOrMethodParams, methodParamsOrDone, _callback);
+      }, (r: any, m: IncomingMessage) => { return createResultWithIncomingMessage(r, m); }, callback);
+    } else {
+      return tripleValueCallbackToPromise((_callback) => {
+        this._invokeDeviceMethod(deviceId, moduleIdOrMethodParams, _callback);
+      }, (r: any, m: IncomingMessage) => { return createResultWithIncomingMessage(r, m); }, callback);
+    }
```

Comment:
Hey @pierreca, sorry I missed that PR. Is other methods when callback can be passed in more than one parameter I tried to "normalize" it before passing to the *callbackToPromise methods (like [here](https://github.com/Azure/azure-iot-sdk-node/blob/8b7cb28a8d8664f63378393e4febb79480d24ed4/service/src/query.ts#L102)). This may be easier to test because we can extract the code getting callback from the list of parameters and test it in isolation.